gen/list_test.go

```diff
@@ -0,0 +1,202 @@
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package gen
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	tc "go.uber.org/thriftrw/gen/internal/tests/containers"
+	"go.uber.org/thriftrw/wire"
+)
+
+// Testing the following contract.
+//
+// ToWire
+//  Required list field
+//      list is nil: should encode into a wire.Value that holds an empty list.
+//      list is empty: should also encode into a wire.Value that holds an empty list.
+//
+//  Optional list field
+//      list is nil: should encode into a wire.Value that does not hold the list field.
+//      list is empty: should encode into a wire.Value that holds an empty list.
+//
+// FromWire
+//  Required list field
+//      wire.Value has an empty list field: decodes into empty.
+//      wire.Value is missing the list field: fails.
+//
+//  Optional list field
+//      wire.Value has an empty list field: decodes into empty.
+//      wire.Value is missing the list field: decodes to nil.
+//
+// We also test that full round trip works as expected with protocol en/decoding.
+
+// TestListRequired tests that empty slice and nil allocate an empty list after deserialization via
+// FromWire.
+func TestListRequiredToWire(t *testing.T) {
+	tests := []struct {
+		desc     string
+		give     *tc.ListOfRequiredPrimitives
+		want     wire.Value
+		wantList []string
+	}{
+		{
+			desc: "required nil list: wire.Value with empty list",
+			give: &tc.ListOfRequiredPrimitives{
+				ListOfStrings: nil,
+			},
+			want: wire.NewValueStruct(wire.Struct{Fields: []wire.Field{
+				{
+					ID:    1,
+					Value: wire.NewValueList(wire.ValueListFromSlice(wire.TBinary, []wire.Value{})),
+				},
+			}}),
+			wantList: []string{}, // we allocate a new slice if it doesn't exist.
+		},
+		{
+			desc: "required empty list: wire.Value with empty list",
+			give: &tc.ListOfRequiredPrimitives{
+				ListOfStrings: []string{},
+			},
+			want: wire.NewValueStruct(wire.Struct{Fields: []wire.Field{
+				{
+					ID:    1,
+					Value: wire.NewValueList(wire.ValueListFromSlice(wire.TBinary, []wire.Value{})),
+				},
+			}}),
+			wantList: []string{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			w, err := tt.give.ToWire()
+			require.NoError(t, err, "failed to serialize: %encodedValue", tt.give)
```

Comment:
Typo: `%encodedValue` isn't a valid formatting argument.

```suggestion
			require.NoError(t, err, "failed to serialize", tt.give)
```